fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/adapters/EventExpandableListViewAdapter.java

```diff
@@ -242,15 +248,57 @@ public View getChildView(
     // in the future it could be nice to have a pencil icon to allow organizer to modify an event
 
     LayoutEventBinding binding;
-    Event event = ((Event) getChild(groupPosition, childPosition));
+
     if (convertView == null) {
       LayoutInflater inflater = LayoutInflater.from(parent.getContext());
       binding = LayoutEventBinding.inflate(inflater, parent, false);
     } else {
       binding = DataBindingUtil.getBinding(convertView);
     }
 
-    binding.setEvent(event);
+    //Event event = ((Event) getChild(groupPosition, childPosition));
+    Event event = (Event)getChild(groupPosition, childPosition);
+    if(event instanceof RollCall){
+      RollCall rollCall = (RollCall)event;
+      binding.setRollCall(rollCall);
+      binding.setIsRollCall(true);
+      SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("HH:mm", Locale.FRENCH);
+      binding.rollCallTime.setText("Time: "+DATE_FORMAT.format(new Date(1000*rollCall.getStart())));
+
+      binding.rollCallTitle.setText("Roll Call: "+rollCall.getName());
+
+      binding.setViewModel(viewModel);
+
+      boolean isOrganizer = viewModel.isOrganizer().getValue();
```

Comment:
I think you should create a separate layout in xml for the Roll Calls , because in this layout you use a lot of attributes juste to know if it's a RollCall and it's a bit messy especially when there will more Events to use the same layout for all the events

Reply:
Thanks for the idea, actually this layout should be similar to all events. The fields displayed are the same and there might only be tiny differences for the buttons. So I will remove the checks you mentioned and rename the elements with "event" instead of "roll_call".